### PR TITLE
feat(dlp): added a sample for inspect string with exclusion dictionary 

### PR DESCRIPTION
**Pull request by AarshD-crest**
- Added a sample for [inspect string with exclusion dictionary](https://cloud.google.com/dlp/docs/samples/dlp-inspect-string-with-exclusion-dict) with unit test-case.

### CODE REVIEW

File: dlp/snippets/inspect/inspect_string_with_exclusion_dict.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Closing the client safely cleans up background resources.
+	defer client.Close()
+
+	// Specify the type and content to be inspected.
+	var contentItem = &dlppb.ContentItem{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_with_exclusion_dict.go:49]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_string_with_exclusion_dict.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+			ByteItem: &dlppb.ByteContentItem{
+				Type: dlppb.ByteContentItem_TEXT_UTF8,
+				Data: []byte(textToInspect),
+			},
+		},
+	}
+
+	// Specify the type of info the inspection will look for.
+	// See https://cloud.google.com/dlp/docs/infotypes-reference for complete list of info types.
+	var infoTypes = []*dlppb.InfoType{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_with_exclusion_dict.go:60]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_string_with_exclusion_dict.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+		{Name: "PHONE_NUMBER"},
+		{Name: "EMAIL_ADDRESS"},
+		{Name: "CREDIT_CARD_NUMBER"},
+	}
+
+	// Exclude matches from the specified excludedMatchList.
+	var exclusionRule = &dlppb.ExclusionRule{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_with_exclusion_dict.go:67]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_string_with_exclusion_dict.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+			RuleSet: []*dlppb.InspectionRuleSet{
+				ruleSet,
+			},
+		},
+	}
+
+	// Send the request.
+	resp, err := client.InspectContent(ctx, req)
+	if err != nil {
+		fmt.Fprintf(w, "Receive: %v", err)
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_with_exclusion_dict.go:111]: issue: just return error to caller.

File: dlp/snippets/inspect/inspect_string_with_exclusion_dict.go (hunk 1 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_string_with_exclusion_dict.go]: fixed it.

File: dlp/snippets/inspect/inspect_string_with_exclusion_dict.go (hunk 3 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_string_with_exclusion_dict.go]: fixed it.

File: dlp/snippets/inspect/inspect_string_with_exclusion_dict.go (hunk 2 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_string_with_exclusion_dict.go]: fixed it.

File: dlp/snippets/inspect/inspect_string_with_exclusion_dict.go [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+						Words: excludedMatchList,
+					},
+				},
+			},
+		},
+		MatchingType: dlppb.MatchingType_MATCHING_TYPE_FULL_MATCH,
+	}
+
+	// Construct a ruleset that applies the exclusion rule to the EMAIL_ADDRESSES infoType.
+	var ruleSet = &dlppb.InspectionRuleSet{
```
[the remark this answers is not in the document]
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_string_with_exclusion_dict.go]: fixed it.

**Review by telpirion — APPROVED** [review]

**Pull request merged**